src/components/settings/SettingsLMSTab/index.jsx

```diff
@@ -111,6 +119,8 @@ export default function SettingsLMSTab(enterpriseId) {
   useEffect(() => {
     // On load fetch potential existing configs
     fetchExistingConfigs();
+    // Check if the customer needs an idp config
```

Comment:
I think it's better to have this set in a separate useEffect of its own. That is because this has a dependency on the values like this, whereas the original useEffect should only ever run once when this component first renders

so I would suggest a separate useEffect like this

```
useEffect(() => {
    // Check if the customer needs an idp config
    setDisplayNeedsSSOAlert(enableSamlConfigurationScreen && !identityProvider);
}, [enableSamlConfigurationScreen, identityProvider] );
```

Reply:
an alternative way to achieve this, perhaps a bit cleaner even, is 

```
const displayNeedsSSOAlert = useMemo( () => enableSamlConfigurationScreen && !identityProvider, [enableSamlConfigurationScreen, identityProvider]);
```

then you won't need the set at all, it's just a regular constant passed down to other components as props. but either option is fine!